.github/workflows/ci.yml

```diff
@@ -58,9 +58,34 @@ jobs:
             run: |
                 make clean local-deb-package local-rpm-package local-apk-package
           - name: Upload Artifacts
-            uses: actions/upload-artifact@v3
+            uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
             with:
-              name: nginx-agent-snapshots
+              name: nginx-agent-unsigned-snapshots
               path: build
-              retention-days: 3
-
+              retention-days: 1
+  integration-tests:
+    name: Integration Tests
+    needs: build-unsigned-snapshot
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        container:
+        - image: "ubuntu"
+          version: "22.04"
+        - image: "redhatenterprise"
+          version: "9"
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version-file: 'go.mod'
+      - name: Download Packages
+        uses: actions/download-artifact@eaceaf801fd36c7dee90939fad912460b18a1ffe # v4.1.2
+        with:
+          name: nginx-agent-unsigned-snapshots
+          path: build
+      - name: Run Integration Tests
+        run: |
+          go install github.com/goreleaser/nfpm/v2/cmd/nfpm@${{ env.NFPM_VERSION }}
+          OS_RELEASE="${{ matrix.container.image }}" OS_VERSION="${{ matrix.container.version }}" \
+            make integration-test
```

Comment:
Would it be worth capturing the logs if they fail like we have done previously https://github.com/nginx/agent/blob/82c7a23f0a0a70e8f91e2bf3a114aee155082c8f/.github/workflows/ci.yml#L160

Reply:
Not sure if we still need to do this anymore. In the test teardown function I log the container logs now every time.
https://github.com/nginx/agent/blob/add-integration-tests/test/integration/dataplane_api_test.go#L63

Reply:
Aww right perfect, thought that was only the agent logs and not all of the logs misread it.